src/zerocash/PourTransaction.h

```diff
@@ -179,18 +184,19 @@ friend class PourProver;
 
 private:
 
-    std::vector<unsigned char>  publicOldValue;      // public input value of the Pour transaction
-    std::vector<unsigned char>  publicNewValue;     // public output value of the Pour transaction
-    std::vector<unsigned char>  serialNumber_1;     // serial number of input (old) coin #1
-    std::vector<unsigned char>  serialNumber_2;     // serial number of input (old) coin #1
-    CoinCommitment              cm_1;               // coin commitment for output coin #1
-    CoinCommitment              cm_2;               // coin commitment for output coin #2
-    std::vector<unsigned char>  MAC_1;              // first MAC    (h_1 in paper notation)
-    std::vector<unsigned char>  MAC_2;              // second MAC   (h_2 in paper notation)
-    std::string                 ciphertext_1;       // ciphertext #1
-    std::string                 ciphertext_2;       // ciphertext #2
-    std::string                 zkSNARK;            // contents of the zkSNARK proof itself
-    uint16_t                    version;            // version for the Pour transaction
+    std::vector<unsigned char>   publicOldValue;      // public input value of the Pour transaction
+    std::vector<unsigned char>   publicNewValue;     // public output value of the Pour transaction
+    std::vector<unsigned char>   serialNumber_1;     // serial number of input (old) coin #1
+    std::vector<unsigned char>   serialNumber_2;     // serial number of input (old) coin #1
+    CoinCommitment               cm_1;               // coin commitment for output coin #1
+    CoinCommitment               cm_2;               // coin commitment for output coin #2
+    std::vector<unsigned char>   MAC_1;              // first MAC    (h_1 in paper notation)
+    std::vector<unsigned char>   MAC_2;              // second MAC   (h_2 in paper notation)
+    ZCNoteEncryption::Ciphertext ciphertext_1;       // ciphertext #1
+    ZCNoteEncryption::Ciphertext ciphertext_2;       // ciphertext #2
+    uint256                      ephemeralKey;       // epk
+    std::string                  zkSNARK;            // contents of the zkSNARK proof itself
+    uint16_t                     version;            // version for the Pour transaction
```

Comment:
Should we use `array` instead of `vector` for all of the above since they're all fixed-length?